File: cpp/16.最接近的三数之和.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) { 
        sort(nums.begin(),nums.end());
        int res=nums[0]+nums[1]+nums.back();
        for(int i=0;i<nums.size()-2;++i)
        {
            int l=i+1,r=nums.size()-1;
            while(l<r)
            {
                int temp=nums[i]+nums[l]+nums[r];
                if(abs(temp-target)<abs(res-target)) res=temp;
                
                if(temp<target) l++;
                else r--;
            }
        }
        return res;
    }
};

File: javascript/16.最接近的三数之和.js
/*
 * @lc app=leetcode.cn id=16 lang=javascript
 *
 * [16] 最接近的三数之和
 */
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    var result = nums[0] + nums[1] + nums[nums.length-1];
    nums.sort((a, b) => a - b);
    for(var i = 0; i < nums.length - 2; i++){
        var start = i + 1, end = nums.length - 1;
        while(start < end){
            var temp = nums[start] + nums[end] + nums[i];
            if(Math.abs(temp - target) < Math.abs(result - target)){
                result = temp;
            }

            if(temp < target){
                start++; 
            }else{
                end--;
            }
        } 
    }
    return result;
};

